lib/internal/module.js

```diff
@@ -83,43 +83,50 @@ const builtinLibs = [
   'string_decoder', 'tls', 'tty', 'url', 'util', 'v8', 'vm', 'zlib'
 ];
 
+// Make built-in modules available directly (loaded lazily).
 function addBuiltinLibsToObject(object) {
-  // Make built-in modules available directly (loaded lazily).
-  builtinLibs.forEach((name) => {
+  const configurable = true;
+  const enumerable = false;
+
+  // Setter function that will be bound to some lib name
+  function setReal(libname, value) {
+    // Deleting the property before re-assigning it disables the
+    // getter/setter mechanism.
+    delete object[libname];
+    object[libname] = value;
+  }
+
+  for (var n = 0, len = builtinLibs.length; n < len; n++) {
     // Goals of this mechanism are:
     // - Lazy loading of built-in modules
     // - Having all built-in modules available as non-enumerable properties
     // - Allowing the user to re-assign these variables as if there were no
     //   pre-existing globals with the same name.
+    const libname = builtinLibs[n];
+    const set = setReal.bind(null, libname);
```

Comment:
Why `bind()` this method but not the getter? I think we should be consistent. I do not know offhand if either method is more efficient, but since there are not so many built-in modules it *may not* make much difference.